invisible_cities/evm/pmaps_test.py

```diff
@@ -305,3 +315,58 @@ def test_PMap_s2s(pmps):
     assert len(pmp.s2s) == len(s2s)
     for kept_s2, true_s2 in zip(pmp.s2s, s2s):
         assert_Peak_equality(kept_s2, true_s2)
+
+
+
+@fixture(scope='module')
+def noise_func():
+    return NoiseSampler('new', 6400).signal_to_noise
+
+
+@fixture(scope='module')
+def s2_peak():
+    times      = np.arange(0, 20 * units.mus, 1 * units.mus)
+    bin_widths = np.full(len(times), 1 * units.mus)
+
+    pmt_ids  =    DB.DataPMT ('new', 6400).SensorID.values
+    n_sipms = len(DB.DataSiPM('new', 6400).SensorID.values)
+    sipm_ids = np.arange(n_sipms)
+
+    pmts  = PMTResponses(pmt_ids,
+                         np.random.uniform(0, 100,
+                                           (len(pmt_ids), len(times))))
+
+    sipms = SiPMResponses(sipm_ids,
+                          np.random.uniform(0, 10,
+                                            (len(sipm_ids), len(times))))
+
+    return S2(times, bin_widths, pmts, sipms)
+
+
+@mark.parametrize("charge_type",
+                  (SiPMCharge.raw, SiPMCharge.sn))
+def test_sipm_charge_array(charge_type,
+                           s2_peak    ,
+                           noise_func ):
+    charge_arr = s2_peak.sipm_charge_array(noise_func ,
+                                           charge_type,
+                                           False      )
+
+    all_wf = s2_peak.sipms.all_waveforms
+    assert np.array(charge_arr).shape[0] == all_wf.shape[1]
+    assert np.array(charge_arr).shape[1] == all_wf.shape[0]
```

Comment:
do you need these `np.array` conversion? If not remove it, if you do maybe convert it in the definition to remove a bit of noise here.

Reply:
also you could just do `assert charge_arr.shape == all_wf.T.shape`

Reply:
I need the np.array because of the way I wrote the function which might be an indication of its suboptimal style. In raw mode it just returns the transpose (essentially) so it wouldn't be necessary but in signal_to_noise mode it's a tuple basically because it seemed like a waste to convert to an array for the uses it gets in the reconstruction city (it just gets looped over)

it returns tuple(map()) which would have to be np.array(tuple(map())) (as far as I know) to avoid this line in the test but would add another operation when running over data. An extra operation seemed more acceptable in a test than in the other case

Reply:
Yes, I agree. I was asking that, in that case, you move the conversion to a previous line (in the test) so it doesn't add noise to the assertion. That may be me being too picky, but I like the assertions to be as clean and clear as possible, as they are usually the lines that define what a test aims to do.